test/expect-tests/dune_lang/sexp_tests.ml

```diff
@@ -353,3 +353,37 @@ comment)
 block
 comment|#
 |}
+
+let print_shell_spec : Dune_lang.Shell_spec.t -> unit =
+ fun spec -> Dune_lang.pp (Dune_lang.Shell_spec.encode spec) |> print
+
+let shell_path = Dune_lang.String_with_vars.make_text loc "/bin/zsh"
+
+let%expect_test _ =
+  print_shell_spec System_shell;
+  [%expect {| :system |}]
+
+let%expect_test _ =
+  print_shell_spec Bash_shell;
+  [%expect {| :bash |}]
+
+let%expect_test _ =
+  print_shell_spec (Exec_file_shell shell_path);
+  [%expect {| /bin/zsh |}]
+
+let print_action : Dune_lang.Action.t -> unit =
+ fun spec -> Dune_lang.pp (Dune_lang.Action.encode spec) |> print
+
+let cram_script = Dune_lang.String_with_vars.make_text loc "test.t"
+
+let%expect_test _ =
+  print_action (Cram (cram_script, System_shell));
+  [%expect {| (cram test.t) |}]
+
+let%expect_test _ =
+  print_action (Cram (cram_script, Bash_shell));
+  [%expect {| (cram test.t (shell :bash)) |}]
```

Comment:
This optional argument of the cram action will need some thinking. We don't have optional arguments in any other user action apart from ones that take lists like `progn`, `concurrency` or `cat`.

Reply:
I would be interested to hear @emillon or @rgrinberg's opinions on this however.

Reply:
How about you just introduce a different action for a cram action that uses a different shell? E.g. `cram_with_shell`?